app/src/main/java/com/example/characters/fragment/UserDetailsFragment.kt

```diff
@@ -1,28 +1,30 @@
 package com.example.characters.fragment
 
+import android.app.AlertDialog
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.Toast
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.lifecycleScope
 import androidx.navigation.fragment.findNavController
 import androidx.navigation.fragment.navArgs
 import coil.load
+import com.example.characters.R
 import com.example.characters.databinding.FragmentUserDetailsBinding
-import com.example.characters.model.UserDetails
-import com.example.characters.retrofit.RetrofitService
-import retrofit2.Call
-import retrofit2.Callback
-import retrofit2.HttpException
-import retrofit2.Response
+import com.example.characters.model.UserDetailsViewModel
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class UserDetailsFragment : Fragment() {
 
     private var _binding: FragmentUserDetailsBinding? = null
     private val binding get() = requireNotNull(_binding)
+
+    private val viewModel by viewModel<UserDetailsViewModel>()
```

Comment:
сюда передавать аргумент из бандла
```suggestion
    private val viewModel by viewModel<UserDetailsViewModel> {
        parametersOf(args.userId)
    }
```